src/components/fx/layout_defaults.js

```diff
@@ -17,6 +17,7 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
     }
 
     coerce('dragmode');
+    coerce('selectdirection');
```

Comment:
This only has an effect for `dragmode: 'select'`, correct? So this be:

```js
var dragMode = coerce('dragmode');
if(dragMode === 'select') {
  coerce('selectdirection')
}
```

Reply:
yep -> conditional coerce https://github.com/plotly/plotly.js/pull/2506/commits/9dd166c69bd654bd0a70dac6d2f0cb4f3d6c7235